iceoryx_posh/test/fuzztests/fuzzing.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "fuzzing.hpp"
+#include "cpptoml.h"
+#include "iceoryx_hoofs/platform/socket.hpp"
+#include "iceoryx_hoofs/posix_wrapper/posix_call.hpp"
+
+#include <fstream>
+#include <thread>
+
+constexpr char UDS_NAME[] = "/tmp/";
+
+void Fuzzing::fuzzingRouDiCom(const std::shared_ptr<RouDiFuzz> aRouDi, const std::string aMessage) noexcept
+{
+    if (aRouDi != nullptr)
```

Comment:
To avoid to perform error handling here I recommend that you change the argument type of `aRouDi` from `const std::shared_ptr<RouDiFuzz>` to `const RouDiFuzz & aRoudi`.

With this it is guaranteed that you get the reference to a valid roudi instance and can get rid of the if and the error handling.